File: 06Adapter/Students/Szarafinski/TwoWayAdapter/AirCraft.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package AdapterTwoWay;

/**
 *
 * @author KrzysieK
 */
public class AirCraft implements IAirCraft {
    int height;
    boolean airborne;
    
    AirCraft(){
        height = 0;
        airborne = false;
    }
           
    @Override
    public Boolean Airborne() {
        return airborne;
    }

    @Override
    public void TakeOff() {
        System.out.println("Aircraft engne take off");
        airborne = true;
        height = 200;
    }

    @Override
    public int Height() {
        return height;
    }
    
}

File: 06Adapter/Students/2018/artkolinski/06Adapter/Seabird/Aircraft.cs
using System;

namespace Seabird
{
    // Target
    public class Aircraft : IAircraft
    {
        int height;
        bool airborne;
        public Aircraft()
        {
            height = 0;
            airborne = false;
        }
        public virtual void TakeOff()
        {
            Console.WriteLine("Aircraft engine takeoff");
            airborne = true;
            height = 200; // Meters
        }
        public virtual bool Airborne
        {
            get { return airborne; }
        }
        public int Height
        {
            get { return height; }
            protected set { height = value; }
        }
    }
}
